pokemongo_bot/__init__.py

```diff
@@ -1,120 +1,113 @@
 # -*- coding: utf-8 -*-
 
-import logging
+import datetime
 import googlemaps
 import json
+import logger
+import logging
 import random
-import threading
-import datetime
 import sys
+import threading
 import yaml
-import logger
-from pgoapi import PGoApi
+
 from cell_workers import PokemonCatchWorker, SeenFortWorker
-from cell_workers.utils import distance
+from cell_workers.utils import filtered_forts, distance
 from human_behaviour import sleep
+from item_list import Item
+from pgoapi import PGoApi
 from stepper import Stepper
+
 from geopy.geocoders import GoogleV3
 from math import radians, sqrt, sin, cos, atan2
-from item_list import Item
 
 
 class PokemonGoBot(object):
+    process_ignored_pokemon = False
 
     def __init__(self, config):
         self.config = config
-        self.pokemon_list = json.load(open('data/pokemon.json'))
         self.item_list = json.load(open('data/items.json'))
+        self.pokemon_list = json.load(open('data/pokemon.json'))
 
     def start(self):
         self._setup_logging()
         self._setup_api()
+        self._setup_ignored_pokemon()
         self.stepper = Stepper(self)
         random.seed()
 
     def take_step(self):
         self.stepper.take_step()
 
-    def work_on_cell(self, cell, position, include_fort_on_path):
-        process_ignore = False
-        try:
-            with open("./data/catch-ignore.yml", 'r') as y:
-                ignores = yaml.load(y)['ignore']
-                if len(ignores) > 0:
-                    process_ignore = True
-        except Exception, e:
-            pass
+    def work_on_cell(self, map_cells, position, include_fort_on_path):
+        self._remove_ignored_pokemon(map_cells)
+
+        if (self.config.mode == "all" or self.config.mode == "poke"):
```

Comment:
could we make this and the lines below on if statement? Or is there a reason to not do so?